vkrunner/vr-test.c

```diff
@@ -383,30 +383,46 @@ bind_ubo_descriptor_set(struct test_data *data)
         struct vr_vk *vkfn = &data->window->vkfn;
         const struct vr_context *context = data->window->context;
 
-        if (data->ubo_descriptor_set_bound ||
-            data->ubo_descriptor_set == VK_NULL_HANDLE)
+        if (data->ubo_descriptor_set_bound || !data->ubo_descriptor_set)
                 return;
 
         if (data->pipeline->stages & ~VK_SHADER_STAGE_COMPUTE_BIT) {
-                vkfn->vkCmdBindDescriptorSets(context->command_buffer,
-                                              VK_PIPELINE_BIND_POINT_GRAPHICS,
-                                              data->pipeline->layout,
-                                              0, /* firstSet */
-                                              1, /* descriptorSetCount */
-                                              &data->ubo_descriptor_set,
-                                              0, /* dynamicOffsetCount */
-                                              NULL /* pDynamicOffsets */);
+                for (unsigned i = 0; i < data->pipeline->n_desc_sets; i++) {
+                        if (data->ubo_descriptor_set[i] == VK_NULL_HANDLE) {
+                                vr_error_message(data->window->config,
+                                                 "Warning buffer descriptor set"
+                                                 " [%u] is VK_NULL_HANDLE",
+                                                 i);
+                                continue;
+                        }
```

Comment:
This check seems a little unnecessary. If it fails it means something has gone wrong internally in VkRunner, right? Maybe it should be an assert or just remove it entirely.